plugins/inventory/vmware_vm_inventory.py

```diff
@@ -247,88 +336,116 @@ def check_requirements(self):
-
-        if properties is None:
-            properties = ['name']
+        TraversalSpec = vmodl.query.PropertyCollector.TraversalSpec
+        FilterSpec = vmodl.query.PropertyCollector.FilterSpec
+        ObjectSpec = vmodl.query.PropertyCollector.ObjectSpec
+        PropertySpec = vmodl.query.PropertyCollector.PropertySpec
 
-        # Create Container View with default root folder
-        mor = self.content.viewManager.CreateContainerView(root_folder, [vim_type], True)
+        resource_filters = resources or []
+        type_to_name_map = {}
 
-        # Create Traversal spec
-        traversal_spec = vmodl.query.PropertyCollector.TraversalSpec(
-            name="traversal_spec",
-            path='view',
-            skip=False,
-            type=vim.view.ContainerView
-        )
+        def _handle_error(message):
+            if strict:
+                raise AnsibleError(message)
+            else:
+                display.warning(message)
+
+        def get_contents(container, vim_types):
+            return self.content.propertyCollector.RetrieveContents([
+                FilterSpec(
+                    objectSet=[
+                        ObjectSpec(
+                            obj=self.content.viewManager.CreateContainerView(
+                                container, vim_types, True),
+                            skip=False,
+                            selectSet=[TraversalSpec(
+                                type=vim.view.ContainerView, path='view', skip=False)]
+                        )],
+                    propSet=[PropertySpec(type=t, all=False, pathSet=['name']) for t in vim_types],
+                )
+            ])
+
+        def filter_containers(containers, typ, filter_list):
+            if len(filter_list) > 0:
+                objs = []
+                results = []
+                found_filters = {}
+
+                for container in containers:
+                    results.extend(get_contents(container, [typ]))
+
+                for res in results:
+                    if res.propSet[0].val in filter_list:
+                        objs.append(res.obj)
+                        found_filters[res.propSet[0].val] = True
+
+                for fil in filter_list:
+                    if fil not in found_filters:
+                        _handle_error("Unable to find %s %s" % (type_to_name_map[typ], fil))
+
+                return objs
+            return containers
+
+        def build_containers(containers, vim_type, names, filters):
+            filters = filters or []
+            if vim_type:
+                containers = filter_containers(containers, vim_type, names)
+
+            new_containers = []
+            for fil in filters:
+                new_filters = None
+                for k, v in fil.items():
+                    if k == "resources":
+                        new_filters = v
+                    else:
+                        vim_type = getattr(vim, _snake_to_camel(k, True))
```

Comment:
What is the `_snake_to_camel` trying to do here? I'm trying to do a test case of `config.name == "DC0_H0_VM0"`. That yields an attribute error that the object has no "Config" attribute. From other examples, I don't see how it should start with a capital "C"

Reply:
That code for `resrouces` logic. It should not trigger if you're not using it. Can you elaborate on your test case?